### PR TITLE
Bump Clang-Tidy from 10 to 11

### PR DESCRIPTION
**Pull request by yykamei**
> Explain a summary, purpose, or background for this change.

LLVM 11 is publicly available, so I want to bump it in this repository.

https://lists.llvm.org/pipermail/llvm-announce/2020-October/000089.html
https://lists.llvm.org/pipermail/llvm-announce/2021-January/000090.html

> Refer related issues or pull requests (e.g. `Close #123` or `None`).

See https://github.com/sider/runners/issues/2152#issuecomment-793429116

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Updated tools:
 - **RuboCop** 1.10.0 -> 1.11.0 [#2140](https://github.com/sider/runners/pull/2140)
 - **stylelint** 13.11.0 -> 13.12.0 [#2143](https://github.com/sider/runners/pull/2143)
 - **SwiftLint** 0.42.0 -> 0.43.0 [#2133](https://github.com/sider/runners/pull/2133)
+- **Clang-Tidy** 10 -> 11 [#2154](https://github.com/sider/runners/pull/2154)
```
Comment (ybiquitous) [inline CHANGELOG.md:20]: [suggestion] Can you please sort the "Updated tools" list alphabetically?

File: test/smokes/clang_tidy/expectations.rb [hunk 2]
```diff
@@ -1,6 +1,6 @@
 s = Runners::Testing::Smoke
 
-default_version = "10.0.1"
+default_version = /^11\./
```
Comment (ybiquitous) [inline test/smokes/clang_tidy/expectations.rb:3]: [suggestion] I think it is better to set the exact version here to ensure the test correctness.
If the CI will be broken, we just need to modify this file.

**Review by ybiquitous — APPROVED** [review]
LGTM 👍

**Comment by yykamei** [comment]
Thanks.

**Pull request merged**